go-controller/pkg/metrics/metrics.go

```diff
@@ -144,6 +151,110 @@ func coverageShowMetricsUpdater(component string) {
 	}
 }
 
+// registerStopwatchShowMetrics registers stopwatch/show metrics for
+// various components(ovn-northd, ovn-controller, and ovs-vswitchd) with prometheus
+func registerStopwatchShowMetrics(component string, metricNamespace string, metricSubsystem string) {
+	stopwatchShowMetricsMap := componentStopwatchShowMetricsMap[component]
+	for metricName, metricInfo := range stopwatchShowMetricsMap {
+		metricInfo.metric = prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: metricNamespace,
+			Subsystem: metricSubsystem,
+			Name:      metricName,
+			Help:      metricInfo.help,
+		})
+		prometheus.MustRegister(metricInfo.metric)
+	}
+}
+
+// getStopwatchShowOutputMap obtains the stopwatch/show metric values for the specified component.
+func getStopwatchShowOutputMap(component string) (map[string]int, error) {
+	var stdout, stderr string
+	var err error
+
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("recovering from a panic while parsing the stopwatch/show output for %s: %v", component, r)
+		}
+	}()
+
+	switch component {
+	case ovnController:
+		stdout, stderr, err = util.RunOVNControllerAppCtl("stopwatch/show")
+	case ovnNorthd:
+		stdout, stderr, err = util.RunOVNNorthAppCtl("stopwatch/show")
```

Comment:
How often do you run? Do you run at the same time as coverage/show?

Reply:
It runs every 30 seconds and is triggered in a separate go routine (e.g. https://github.com/ovn-org/ovn-kubernetes/blob/8406f70c24b3936dd301a9a781e856a3f0cf8963/go-controller/pkg/metrics/ovn.go#L419). `coverage/show` is handled in another go routine (e.g. https://github.com/ovn-org/ovn-kubernetes/blob/8406f70c24b3936dd301a9a781e856a3f0cf8963/go-controller/pkg/metrics/ovn.go#L417), so they will not run at the same time (in most cases).